### PR TITLE
Support dropout seed offset as pointer

### PR DESCRIPTION
**Pull request by rocking5566**
new CK support dropout seed and offset as pointer

### CODE REVIEW

File: csrc/flash_attn_ck/mha_bwd.cpp [hunk 1]
```diff
@@ -337,21 +336,26 @@ mha_bwd(const at::Tensor &dout,                   // batch_size x seqlen_q x num
...
+        rng_state = torch::empty({2}, opts.dtype(torch::kInt64));
         // See Note [Acquire lock when using random generators]
         std::lock_guard<std::mutex> lock(gen->mutex_);
         auto philox_args = gen->philox_cuda_state(counter_offset);
-        std::tie(drop_seed, drop_offset) = flash::unpack(philox_args);
+        hipLaunchKernelGGL(
+            flash::ParsePhiloxCudaState, dim3(1), dim3(64), 0, 0,
+            philox_args, reinterpret_cast<uint64_t*>(rng_state.data_ptr()));
+    } else {
+        rng_state = torch::empty({2}, opts.dtype(torch::kInt64));
```
Comment (poyenc) [inline csrc/flash_attn_ck/mha_bwd.cpp:353]: is it necessary to add the `else` case?

File: csrc/flash_attn_ck/mha_bwd.cpp [hunk 2]
```diff
@@ -191,7 +190,7 @@ fmha_bwd_args get_ck_fmha_bwd_args(const mask_info &mask,
                          static_cast<ck_tile::index_t>(mask.type),
                          p_dropout,
                          p_undrop,
-                         {drop_seed, drop_offset}};
+                         drop_seed_offset};
```
Comment (poyenc) [inline csrc/flash_attn_ck/mha_bwd.cpp:193]: do we still need the `variant<>` parameter?

File: csrc/flash_attn_ck/mha_bwd.cpp [hunk 3]
```diff
@@ -337,21 +336,26 @@ mha_bwd(const at::Tensor &dout,                   // batch_size x seqlen_q x num
...
-    uint64_t drop_seed = 1, drop_offset = 0;
     int64_t counter_offset = batch_size * num_heads * ck_tile::get_warp_size();
+    at::Tensor rng_state;
 
-    if (rng_state.has_value()) {
-        uint64_t* d = reinterpret_cast<uint64_t*>(rng_state.value().data_ptr());
-        drop_seed = d[0];
-        drop_offset = d[1];
+    if (rng_state_.has_value()) {
+        rng_state = rng_state_.value();
```
[the remark this answers is not in the document]
Comment (rocking5566) [reply csrc/flash_attn_ck/mha_bwd.cpp:343]: Yes

**Review by poyenc — APPROVED** [review]

File: csrc/flash_attn_ck/mha_bwd.cpp (hunk 2 above)
Comment (rocking5566) [reply csrc/flash_attn_ck/mha_bwd.cpp:193]: other user might need value version of dropout seed & offset

**Pull request merged**